### PR TITLE
Mark unused but desired private constructors as such

### PR DESCRIPTION
**Pull request by srawlins**
## Description

The flutter codebase has "unused" but intentional private constructors, in place to prevent instantiation and extension. The analyzer will begin to report unused private constructors. These constructors need to be marked with `// ignore: unused_element`.

## Related Issues

https://github.com/dart-lang/sdk/issues/35108

## Tests

I added no tests.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by srawlins** [comment]
CC @goderbauer

File: examples/flutter_gallery/lib/gallery/icons.dart [hunk 1]
```diff
@@ -5,6 +5,9 @@
 import 'package:flutter/material.dart';
 
 class GalleryIcons {
+  // This class is intended to be used as an interface, and should not be
```
Comment (goderbauer) [inline examples/flutter_gallery/lib/gallery/icons.dart]: This class is not really intended to be used an an interface. It's just a grouping mechanism for a collection of statics. Extending (or implementing it) just doesn't make much sense.

Comment (goderbauer) [reply examples/flutter_gallery/lib/gallery/icons.dart]: Same is true for some of the other classes that only have statics below.

Comment (srawlins) [reply examples/flutter_gallery/lib/gallery/icons.dart]: I believe I fixed all of these. Sorry for the delay.

File: packages/flutter/lib/src/foundation/licenses.dart [hunk 2]
```diff
@@ -288,6 +288,9 @@ class LicenseEntryWithLineBreaks extends LicenseEntry {
 ///  * [AboutListTile], which is a widget that can be added to a [Drawer]. When
 ///    tapped it calls [showAboutDialog].
 class LicenseRegistry {
+  // This class is intended to be used as an interface, and should not be
```
Comment (goderbauer) [inline packages/flutter/lib/src/foundation/licenses.dart]: This one is also not meant to be instantiated.

File: packages/flutter_test/lib/src/test_async_utils.dart [hunk 3]
```diff
@@ -7,6 +7,9 @@ import 'dart:async';
 import 'package:flutter/foundation.dart';
 
 class _AsyncScope {
+  // This class is intended to be used as an interface, and should not be
+  // extended directly; this constructor prevents instantiation and extension.
+  // ignore: unused_element
```
Comment (goderbauer) [inline packages/flutter_test/lib/src/test_async_utils.dart]: Why does this one need to be ignored? It's instantiated in line 70 below.

Comment (srawlins) [reply packages/flutter_test/lib/src/test_async_utils.dart]: Sorry, wrong class in this file. Fixed.

File: packages/flutter/lib/src/foundation/licenses.dart (hunk 2 above)
Comment (srawlins) [reply packages/flutter/lib/src/foundation/licenses.dart]: Done

**Review by goderbauer — APPROVED** [review]
LGTM

**Pull request merged**